src/api/libcellml/entity.h

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2015 University of Auckland
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.Some license of other
+*/
+
+#ifndef LIBCELLML_LIBCELLML_ENTITY_H_
+#define LIBCELLML_LIBCELLML_ENTITY_H_
+
+#include "libcellml/enumerations.h"
+
+namespace libcellml {
+
+/**
+ * @brief The Entity class.
+ * Base class for all serialisable libCellML classes.
+ */
+class Entity
+{
+protected:
+    /**
+     * @brief Pure virtual serialise method.
+     * Pure virtual serialise method for serialising a CellML object to a std::string.
+     * @param format The format to serialise the object to.
+     * @return std::string representation of the object.
+     */
+    virtual std::string serialise(libcellml::CELLML_FORMATS format) const = 0;
```

Comment:
Hugh, I can see your point about passing an enumeration as a ```const```. Still, I do it so so that I don't have to 'worry' about 'special' cases.

Regarding naming conventions for local variables, function parameters and class members, it's really a matter of making the code easier to read. I know that since I have started using a prefix of ```m``` for class members and ```p``` for function parameters, my code has been much easier to read (for me, at least). Just my 2c/2p worth of experience.

However, I wouldn't personally go with the convention used in Chaste (except for class members). So, to sum up, I would use:
 - Local variable: ```myCellmlVariable```
 - Function parameter: ```pMyCellmlParameter```
 - Class member: ```mMyCellmlMember```